app/src/main/java/com/bunbeauty/papakarlo/feature/address/screen/create_address/CreateAddressFragment.kt

```diff
@@ -18,51 +18,98 @@ import androidx.compose.ui.Modifier
+import com.bunbeauty.papakarlo.common.ui.element.LoadingButton
 import com.bunbeauty.papakarlo.common.ui.screen.ErrorScreen
 import com.bunbeauty.papakarlo.common.ui.screen.LoadingScreen
 import com.bunbeauty.papakarlo.common.ui.theme.FoodDeliveryTheme
 import com.bunbeauty.papakarlo.databinding.FragmentCreateAddressBinding
 import com.bunbeauty.papakarlo.extensions.compose
-import com.bunbeauty.papakarlo.feature.address.model.StreetItem
+import com.bunbeauty.papakarlo.extensions.showSnackbar
 import com.bunbeauty.papakarlo.feature.address.ui.auto_complete_text_field.AutoCompleteEditText
 import com.bunbeauty.papakarlo.feature.edit_text.model.EditTextType
+import com.bunbeauty.shared.domain.interactor.address.CreateAddressUseCase
+import com.bunbeauty.shared.domain.interactor.street.GetStreetsUseCase
+import com.bunbeauty.shared.presentation.create_address.CreateAddressState
+import com.bunbeauty.shared.presentation.create_address.CreateAddressViewModel
 import org.koin.androidx.viewmodel.ext.android.viewModel
 
-class CreateAddressFragment : BaseFragment(R.layout.fragment_create_address) {
+class CreateAddressFragment : BaseFragmentWithSharedViewModel(R.layout.fragment_create_address) {
 
-    override val viewModel: CreateAddressViewModel by viewModel()
+    val viewModel: CreateAddressViewModel by viewModel()
     override val viewBinding by viewBinding(FragmentCreateAddressBinding::bind)
 
+    @OptIn(ExperimentalLifecycleComposeApi::class)
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
         viewModel.getStreetList()
         viewBinding.fragmentCreateAddressCvMain.compose {
-            val streetListState by viewModel.streetListState.collectAsState()
+            val streetListState by viewModel.streetListState.collectAsStateWithLifecycle()
             CreateAddressScreen(streetListState)
+            LaunchedEffect(streetListState.eventList) {
+                handleEventList(streetListState.eventList)
+            }
         }
     }
 
     @Composable
-    private fun CreateAddressScreen(streetListState: State<List<StreetItem>>) {
-        when (streetListState) {
-            is State.Success -> CreateAddressSuccessScreen(streetListState.data)
-            is State.Loading -> LoadingScreen()
-            is State.Error -> ErrorScreen(streetListState.message) {
-                viewModel.getStreetList()
+    private fun CreateAddressScreen(createAddressState: CreateAddressState) {
+        when (val state = createAddressState.state) {
+            is CreateAddressState.State.Success -> CreateAddressSuccessScreen(createAddressState)
+            is CreateAddressState.State.Loading -> LoadingScreen()
+            is CreateAddressState.State.Error -> {
+                when (state.throwable) {
+                    is GetStreetsUseCase.NoSelectedCityUuidThrow, CreateAddressUseCase.NoSelectedCityUuidThrow -> {
```

Comment:
It's the same exception, it's better to make it common and not to create it for every use case

Reply:
+